pkg/controller/cache.go

```diff
@@ -142,6 +142,24 @@ func (c *GVKRoutedCache) GetInformerForKind(ctx context.Context, gvk schema.Grou
 	return c.fallback.GetInformerForKind(ctx, gvk, opts...)
 }
 
+// RemoveInformer removes an informer entry and stops it if it was running.
+func (c *GVKRoutedCache) RemoveInformer(ctx context.Context, obj client.Object) error {
+	gvk, err := apiutil.GVKForObject(obj, c.scheme)
+	if err != nil {
+		return errors.Errorf("failed to get GVK for type %T: %w", obj, err)
+	}
+
+	c.lock.RLock()
+	delegate, ok := c.delegates[gvk]
+	c.lock.RUnlock()
```

Comment:
should this remove the delegate from the map?

Reply:
I don't know a way to determine if a delegate has any more informers.